### PR TITLE
Add admin dash

### PR DESCRIPTION
**Pull request by mikeyrayvon**

### CODE REVIEW

File: app/imports/api/usersMethods.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      if (!Meteor.userId()) {
+        throw new Meteor.Error('Users.methods.removeWorker.not-logged-in', 'Must be logged in to remove a Worker.');
+      }
+
+      // Check if user is admin
+      if (!Roles.userIsInRole(Meteor.userId(), 'admin')) {
+        throw new Meteor.Error('Users.methods.removeWorker.not-allowed', 'Must be Admin to do this.');
+      }
+
+      Meteor.users.remove(workerId);
```
Comment (varffs) [inline app/imports/api/usersMethods.js:61]: I think the method should return this not just invoke it. But perhaps doesnt matter

File: app/imports/components/workers/AddWorker.jsx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    this.setState({
+      validationError: undefined,
+    });
+
+    let isValid = true;
+
+    const worker = this.getCurrentCleanState();
+
+    try {
+      this.validate(worker);
```
Comment (varffs) [inline app/imports/components/workers/AddWorker.jsx:93]: how does `validate()` know what is it validating against?

File: app/imports/components/workers/AddWorker.jsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+      }
+    }
+
+    if (isValid) {
+      addWorker.call(worker, (err, res) => {
+        if (err) {
+          console.error(err);
+        } else {
+          this.cleanForm();
+          console.log('success');
```
Comment (varffs) [inline app/imports/components/workers/AddWorker.jsx:108]: do we need this log?

File: app/imports/components/workers/AddWorker.jsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+      validationError: err.reason,
+    });
+  }
+
+  render() {
+    return(
+      <form onSubmit={this.onSubmitHandle}>
+        <div className='grid-row'>
+          <div className='grid-item item-s-12'>
+            <h2>Add Worker</h2>
```
Comment (varffs) [inline app/imports/components/workers/AddWorker.jsx]: I think we should add some basic paddings/margins to the layouts

File: app/imports/components/workers/RemoveWorker.jsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+  onRemoveWorker() {
+    removeWorker.call({
+      workerId: this.props.workerId,
+    });
+  }
+
+  render() {
+    return (
+      <button className="pt-button pt-large" onClick={this.onRemoveWorker}>
```
Comment (varffs) [inline app/imports/components/workers/RemoveWorker.jsx]: I feel like this button needn't be large, but we just need a convention across the code

File: app/imports/containers/ListWorkersContainer.jsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  };
+};
+
+function reactiveMapper(props, onData) {
+
+  if (Meteor.subscribe('users.workers').ready()) {
+
+    const user = !!Meteor.user() ? Meteor.user() : null;
+
+    if (Roles.userIsInRole(user._id, [ 'admin' ])) {
```
Comment (varffs) [inline app/imports/containers/ListWorkersContainer.jsx]: not sure we need the double check as the subscription is already checked

File: app/imports/startup/server/publications.jsx [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import { Meteor } from 'meteor/meteor';
+
+Meteor.publish('users.workers', function() {
+
+  if (Roles.userIsInRole(this.userId, 'admin')) {
```
Comment (varffs) [inline app/imports/startup/server/publications.jsx]: this raises the question of if we should lock down all subscriptions to users/roles. Probably so and therefor the projects list subscription needs that added

**Review by varffs — COMMENTED** [review]
Works well. Just a few general questions and possible some additional work on styling/layout

File: app/imports/components/dashboards/MainDashboard.jsx [hunk 8]
```diff
@@ -1,13 +1,22 @@
...
-        </section>
-      </div>
+      <section id="dashboard">
+        <div className="grid-row">
+          <div className="grid-item item-s-12">
+            <h1>Dashboard</h1>
+          </div>
+        </div>
+
+        <ListWorkersContainer />
```
Comment (varffs) [inline app/imports/components/dashboards/MainDashboard.jsx]: Oh but either this needs to be conditionally shown according to user role or put in its own route which is admin user only. I would vote for own route

**Review by varffs — CHANGES_REQUESTED** [review]
When doing the update from master I realised the big question of where this dashboard goes

File: app/imports/api/usersMethods.js (hunk 1 above)
Comment (mixcas) [reply app/imports/api/usersMethods.js:61]: I think `Method.Error` already gets back to the caller function.

> If you want to return an error from a method, throw an exception. Methods can throw any kind of exception. But Meteor.Error is the only kind of error that a server will send to the client.

File: app/imports/containers/ListWorkersContainer.jsx (hunk 6 above)
Comment (mixcas) [reply app/imports/containers/ListWorkersContainer.jsx]: true

File: app/imports/components/workers/AddWorker.jsx (hunk 2 above)
Comment (mixcas) [reply app/imports/components/workers/AddWorker.jsx:93]: Defined in line 73

File: app/imports/api/usersMethods.js (hunk 1 above)
Comment (mixcas) [reply app/imports/api/usersMethods.js:61]: Actually maybe use this http://smashingthingstogether.com/handling-errors-in-meteor-methods/#the-wrapper-function:a41366158d30f9a95269470fa2f88962

**Comment by varffs** [comment]
As discussed offline actions to take are:

- refine role related conditionals
- add basic layout margins
- move to admin role only route

**Review by varffs — APPROVED** [review]

**Pull request merged**